sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2806,7 +2823,15 @@ def _teardown(self,
             with filelock.FileLock(
                     lock_path,
                     backend_utils.CLUSTER_STATUS_LOCK_TIMEOUT_SECONDS):
-                success = self.teardown_no_lock(handle, terminate, purge)
+                success = self.teardown_no_lock(
+                    handle,
+                    terminate,
+                    purge,
+                    # The refresh code path checks current user identity can
+                    # throw ClusterOwnerIdentityMismatchError. However, the
+                    # argument/flag `purge` should bypass such ID mismatch
+                    # errors.
+                    refresh_cluster_status=not purge)
```

Comment:
Disabling refreshing is not safe, as in that case, we may use `ray down` to terminate a autostopped cluster, which will cause leakage, even if the cluster has the same identity and `purge` is specified.

Reply:
Great catch! I just updated the PR.

However, the latest treatment still has a problem -- for the repro in PR description, it'll proceed to do the status refresh but skip the ID check, but this will return a confusing log (`Cluster 'sso' is already terminated. Skipped.`):
```
» sky down sso -p
Terminating 1 cluster: sso. Proceed? [Y/n]:
Terminating 1 cluster ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% -:--:--E 05-11 22:46:17 cloud_vm_ray_backend.py:2804] 'sso' (AWS) is owned by account ['ABCD:zongheng', '1234'], but the activated account is ['5678', '5678'].
W 05-11 22:46:17 cloud_vm_ray_backend.py:2808] Purge (-p/--purge) is set, ignoring the above error and proceeding to attempt to terminate the cluster.
W 05-11 22:46:17 cloud_vm_ray_backend.py:2808] It is the user's responsibility to ensure that this cluster is actually terminated on the cloud.
Terminating 1 cluster ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% -:--:--
W 05-11 22:46:33 cloud_vm_ray_backend.py:3044] Cluster 'sso' is already terminated. Skipped.
Terminating cluster sso...done.
Terminating 1 cluster ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
```
It's also confusing that we say "proceeding to attempt to terminate the cluster" but we don't actually attempt to terminate it.

I'll think about what's the best thing to do here later, but welcome suggestions :)